it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/pubsub/PubsubResourceManager.java

```diff
@@ -394,6 +400,50 @@ private static <T> RetryPolicy<T> retryOnDeadlineExceeded() {
         .build();
   }
 
+  /**
+   * Collects the performance metrics for the pubsub resources.
+   *
```

Comment:
Don't see any UTs updated. Is it possible to unit test any of this? 

We have a `Map` of metrics, it should be possible to assert on it?

Reply:
I have added UTs for Pubsub and Spanner RMs.